src/rpm/builder.rs

```diff
@@ -746,11 +747,21 @@ impl RPMBuilder {
                 offset,
                 IndexData::I18NString(vec![self.desc]),
             ),
-            IndexEntry::new(
-                IndexTag::RPMTAG_LONGSIZE,
-                offset,
-                IndexData::Int64(vec![combined_file_sizes]),
-            ),
+            if small_package {
+                IndexEntry::new(
+                    IndexTag::RPMTAG_SIZE,
+                    offset,
+                    IndexData::Int32(vec![combined_file_sizes
+                        .try_into()
+                        .expect("combined_file_sizes is smaller than 4 GiB")]),
```

Comment:
Just using `as` would be safe here as you've already checked the size.  If you prefer the falliable methods, I think this message is actually backwards, the value is bigger than expected.

Reply:
Unwrapping provides a safety net in case of potential mistakes, so it's better to keep it.

Reply:
As per [std docs](https://doc.rust-lang.org/std/result/enum.Result.html#recommended-message-style) `expect` message describes the reason we expect the `Result` should be `Ok`. I will update the message to use "should be".